src/transport/raw/MessageHeader.h

```diff
@@ -144,6 +150,12 @@ class PacketHeader
     /** Check if it's a secure session control message. */
     bool IsSecureSessionControlMsg() const { return mFlags.Has(Header::FlagValues::kSecureSessionControlMessage); }
 
+    /** Check if the peer's group key message counter is not synchronized. */
+    bool IsPeerGroupMsgIdNotSynchronized() const { return mFlags.Has(Header::FlagValues::kPeerGroupMsgIdNotSynchronized); }
+
+    /** Check if the message buffer should not be freed after sending. */
+    bool IsRetainedBuffer() const { return mFlags.Has(Header::FlagValues::kRetainedBuffer); }
```

Comment:
Where and why would this get used?  We used to have a thing for this, and we removed it, due to lack of use cases and the caller being able to just hold a handle if that's what they want.

Reply:
We do. https://github.com/project-chip/connectedhomeip/issues/4372

When retaining a buffer with CRMP, the caller expects the msg to be unmodified. LwIP stack will normally prepend the packet headers as the packet traverses the UDP/IP/netif layers, which normally modifies the packet.

We need to deep-copies msg into a fresh object, and queues that for transmission, leaving the original msg available after return.

Reply:
I'll follow up in #4372, but I don't see why that needs this API on the packet header.

Reply:
This flag finally needs to be passed to Inet layer
INET_ERROR UDPEndPoint::SendMsg(const IPPacketInfo * pktInfo, System::PacketBufferHandle msg, uint16_t sendFlags)

The buffer is retained in messaging layer, currently, we only have PacketHeader and PayloadHeader passed to transport layer. Passing this flag from Messaging layer to Transport layer via PacketHeader seems the only viable path.

I will post another PR to pass this flag to Inet::UDP for  #4372